### PR TITLE
fix: replace greedy regex with balanced brace parsing in extractJson

### PR DESCRIPTION
**Pull request by kyu1204**
## Summary
- Replaced greedy regex pattern `/\{[\s\S]*\}/` with balanced brace counter
- Fixes incorrect JSON parsing when text contains multiple objects or nested structures
- Added two test cases for edge cases

## Test plan
- [x] Unit tests for parseNaturalLanguage with multiple JSON objects
- [x] Unit tests for parseNaturalLanguage with nested braces in JSON values
- [x] All 33 nl-parse-intent tests pass
- [x] No TypeScript errors
- [x] Build succeeds

🤖 Generated with Claude Code

## 버그 수정
* 자연어 입력에서 JSON 추출 파싱 알고리즘을 개선하여 복잡한 중첩 구조와 여러 JSON 객체가 포함된 경우를 더 안정적으로 처리하도록 변경

## 테스트
* JSON 추출 파싱의 정확성을 검증하기 위한 단위 테스트 추가

### CODE REVIEW

**Comment by kyu1204** [comment]
@coderabbitai 리뷰 진행

**Comment by kyu1204** [comment]
@coderabbitai 리뷰 진행

File: src/nl/parse-intent.ts [hunk 1]
```diff
@@ -39,9 +39,20 @@ export const defaultClaudeRunner: LLMRunner = async (prompt) => {
...
+  let depth = 0;
+  for (let i = start; i < text.length; i++) {
+    if (text[i] === "{") depth++;
+    else if (text[i] === "}") {
+      depth--;
+      if (depth === 0) return text.slice(start, i + 1);
+    }
+  }
+  // If no balanced closing brace found, return the rest of the text
   return text.trim();
```
[the remark this answers is not in the document]
Comment (kyu1204) [reply src/nl/parse-intent.ts]: 반영했습니다. commit a7039ff에서 extractJson이 문자열/escape 상태를 추적하도록 수정했고, quoted braces / escaped quotes / unbalanced fallback 케이스를 tests/unit/nl-parse-intent.test.ts에 추가했습니다.

[the remark this answers is not in the document]
Comment (kyu1204) [reply src/nl/parse-intent.ts]: Addressed in a7039ff. `extractJson()` now tracks string/escape state while scanning braces and falls back to `text.slice(start)` when the JSON stays unbalanced. Focused tests cover quoted braces, escaped quotes, and malformed trailing input.
